### PR TITLE
feat: citation search in Missions sidebar and tally load (#301)

### PR DESCRIPTION
**Pull request by ksalamy**
## Summary

Adds a **citation search** control to the Missions filter sidebar and extends the survey tally **load pipeline** for citation columns, with tests.

Closes mbari-org/SeafloorMappingDB#301

## Changes

- **Filters:** `citation_search` on `MissionFilter` — matches missions whose linked citations contain the search text in DOI or full reference (`icontains`).
- **Load:** Ingest `Citation_1` / `Citation_2` legacy tally columns and the newer `Citations` column format (`scripts/load.py`).
- **Model / DB:** Migration and `Citation` field updates as in diff.
- **UI:** Sidebar / JS updates for the filter (`map.js`, `project.js`).
- **Tests:** `smdb/tests/test_load.py` — citation loading and edge cases.
- **Compose:** `production_localhost.yml` — env-driven paths where applicable.

## Testing

```bash
docker exec django bash -c "cd /app/smdb && DJANGO_SETTINGS_MODULE=config.settings.test pytest smdb/tests/test_load.py -v"
docker exec django bash -c "cd /app/smdb && DJANGO_SETTINGS_MODULE=config.settings.test pytest -v"
```

## Review

Please run **GitHub Copilot** (or requested reviewers) on this PR before merge.

### CODE REVIEW

File: smdb/smdb/filters.py [hunk 1]
```diff
@@ -167,11 +159,12 @@ class Meta:
...
     def filter_citation_search(queryset, name, value):
-        """Filter missions that have at least one citation matching DOI or full_reference (icontains)."""
+        """Filter missions whose linked citations contain value in DOI or full reference."""
         if not value or not value.strip():
             return queryset
         return queryset.filter(
-            Q(citations__doi__icontains=value) | Q(citations__full_reference__icontains=value)
+            Q(citations__doi__icontains=value)
+            | Q(citations__full_reference__icontains=value)
         ).distinct()
```
[the remark this answers is not in the document]
Comment (ksalamy) [reply smdb/smdb/filters.py:168]: Strip + legacy NaN are already handled; wide form still includes citation; sidebar is search-only by design; added test_mission_filter_by_citation_search for regression coverage.
